### PR TITLE
New random event: Wisdom cow

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
add: A wise cow can be found visiting the station sometimes now. Be sure to ask him for advice
/:cl:

he grants you knowledge on a random skill. thanks cow.

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
based 🐄by moo

**Comment by Whoneedspacee** [comment]
runescape genie turned into a cow

File: code/modules/events/wisdomcow.dm [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/datum/round_event_control/wisdomcow
+	name = "Wisdom cow"
+	typepath = /datum/round_event/wisdomcow
+	max_occurrences = 1
+	weight = 20
+
+/datum/round_event/wisdomcow/announce(fake)
+	priority_announce("A wise cow has been spotted in the area. Be sure to ask his advice.", "Nanotrasen Cow Ranching Agency")
```
Comment (ATH1909) [inline code/modules/events/wisdomcow.dm]: are you sure it's a "he"?

wouldn't a male cow be a steer, anyway?

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 2]
```diff
@@ -202,6 +202,31 @@
 	else
 		..()
 
+///Wisdom cow, gives XP to a random skill and speaks wisdoms
+/mob/living/simple_animal/cow/wisdom
+	name = "wisdom cow"
+	desc = "Known for its wisdom, shares it with all"
+	gold_core_spawnable = FALSE
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: :(

**Comment by ATH1909** [comment]
wisdom cow wisdom cow

File: code/modules/events/wisdomcow.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/events/wisdomcow.dm]: wisdom bull doesn't ring off the tongue

File: code/modules/events/wisdomcow.dm [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/datum/round_event_control/wisdomcow
+	name = "Wisdom cow"
+	typepath = /datum/round_event/wisdomcow
+	max_occurrences = 1
+	weight = 20
```
Comment (Time-Green) [inline code/modules/events/wisdomcow.dm]: That's as common as an anomaly, are you sure about that?

Comment (Qustinnus) [reply code/modules/events/wisdomcow.dm]: its as common as a lot of very low impact events

**Comment by optimumtact** [comment]
```
 _____________________________________
/ People never lie so much as after a \
| hunt, during a war, or before an    |
| election.                           |
|                                     |
\ -- Otto Von Bismarck                /
 -------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||
```

**Comment by PKPenguin321** [comment]
for me it's xcowsay fortune

File: code/modules/events/wisdomcow.dm [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+/datum/round_event/wisdomcow/announce(fake)
+	priority_announce("A wise cow has been spotted in the area. Be sure to ask his advice.", "Nanotrasen Cow Ranching Agency")
+
+/datum/round_event/wisdomcow/start()
+	var/turf/targetloc = get_random_station_turf()
+	new /mob/living/simple_animal/cow/wisdom(targetloc)
+	var/datum/effect_system/smoke_spread/smoke = new
+	smoke.set_up(1, targetloc)
+	smoke.start()
```
Comment (Cyberboss) [inline code/modules/events/wisdomcow.dm]: Don't we have something like `do_sparks()` for smoke?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/events/wisdomcow.dm (hunk 1 above)
Comment (optimumtact) [reply code/modules/events/wisdomcow.dm]: yeah it would be her

Comment (Qustinnus) [reply code/modules/events/wisdomcow.dm]: its 2020 let them identify as they want to

jk fixed it

File: code/modules/events/wisdomcow.dm (hunk 4 above)
Comment (Qustinnus) [reply code/modules/events/wisdomcow.dm]: not as far as I know, I looked at how other things do it and this seems like the most common way.

Comment (Qustinnus) [reply code/modules/events/wisdomcow.dm]: Apparently we do, I'll throw that in tonight

File: code/modules/events/wisdomcow.dm [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/datum/round_event_control/wisdomcow
+	name = "Wisdom cow"
+	typepath = /datum/round_event/wisdomcow
+	max_occurrences = 1
+	weight = 20
+
+/datum/round_event/wisdomcow/announce(fake)
+	priority_announce("A wise cow has been spotted in the area. Be sure to ask her advice.", "Nanotrasen Cow Ranching Agency")
```
Comment (Fikou) [inline code/modules/events/wisdomcow.dm]: shouldn't it be ask for her advice or ask her for advice

Comment (ninjanomnom) [reply code/modules/events/wisdomcow.dm]: This would be correct

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 6]
```diff
@@ -202,6 +202,31 @@
...
+
+/mob/living/simple_animal/cow/wisdom/Initialize()
+	. = ..()
+	speak = GLOB.wisdoms //Done here so it's setup properly
+
+///Give intense wisdom to the attacker if they're being friendly about it
+/mob/living/simple_animal/cow/wisdom/attack_hand(mob/living/carbon/M)
+	if(!stat && M.a_intent == INTENT_HELP)
+		to_chat(M, "<span class='nicegreen'>[src] whispers you some intense wisdoms and then dissapears!</span>")
+		M.mind?.adjust_experience(pick(subtypesof(/datum/skill)), 500)
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: The problem is there's no shared scale of experience between different skills iirc, you may be better off increasing their experience in a skill based on a percentage of their current skill. Increase their exp by 10% perhaps

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 7]
```diff
@@ -202,6 +202,31 @@
...
+		var/datum/effect_system/smoke_spread/smoke = new
+		smoke.set_up(1, get_turf(src))
+		smoke.start()
+		qdel(src)
+		return
+	..()
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]:
```suggestion
	return ..()

```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by ninjanomnom** [comment]
Are these fortune cookie quotes, these are predictions not wise quotes.

**Comment by Qustinnus** [comment]
They're from fortune, oranges gave me em

**Comment by actioninja** [comment]
Yeah, ok.

**Pull request merged**